pytorch_lightning/utilities/cli.py

```diff
@@ -591,11 +600,24 @@ def setup_parser(
 
     def add_default_arguments_to_parser(self, parser: LightningArgumentParser) -> None:
         """Adds default arguments to the parser."""
+
+        min_value = np.iinfo(np.int32).min
+        max_value = np.iinfo(np.int32).max
+        int32 = restricted_number_type("int32", int, [(">=", min_value), ("<=", max_value)])
+
+        if isinstance(self.seed_everything_default, bool):
+            if self.seed_everything_default:
+                seed = random.randint(min_value, max_value)
+            else:
+                seed = None
+        else:
+            seed = self.seed_everything_default
```

Comment:
To me this creates a weird behavior. When I do `cli.py --help` what I expect to see as default value is `True` and not a number that changes each time it is run. Similarly when I do `--print_config` the output should have the default value `True`.

Only in the case that the code is run, the saved config would include the used seed. As commented in the issue, the value should be overwritten in the config after parsing. Here the default as it was before.